### PR TITLE
Remove Schedulable Condition. Only Unschedulable bit from Spec will be used.

### LINKED ISSUE #6077: Need reconciliation loop for node schedulable
Node controller stops posting node status, which means it also stops posting node schedulable bits.  What this implies is that when user mark a node unscheduable, it will never work, scheduler will still schedule pods on it, because node status is not updated by node controller anymore.

We need to merge status from node controller and kubelet.  node controller still syncs node, only if its observed status changed.

This is somehow similar to #6072

### PR DESCRIPTION
**Pull request by gmarek**
closes #6077

### CODE REVIEW

**Comment by gmarek** [comment]
I don't know what to do with v1beta1 and v1beta2 APIs. Can we remove NodeConditionKind corresponding to NodeSchedulable, or should we fill it up using data from Spec? @davidopp @bgrant0607 @thockin

**Comment by bgrant0607** [comment]
If it's not hard, just fill it from spec. If it is hard, don't remove the string definition, but don't populate the data.

File: docs/node.md [hunk 1]
```diff
@@ -38,24 +38,19 @@ must have appropriate conditions, see below.
...
 Node Condition describes the conditions of `Running` nodes. Current valid
-conditions are `NodeReady` and `NodeSchedulable`. In the future, we plan to
-add more. `NodeReady` means kubelet is healthy and ready to accept pods
-`NodeSchedulable` means node is allowed to schedule any new pods and is
-controlled by 'unschedulable' field in node spec. Different condition provides
-different level of understanding for node health. Kubernetes will make a
-comprehensive scheduling decision based on the information. Node condition
-is represented as a json object. For example, the following conditions mean
-the node is in sane state but not allowed to accept new pods:
+conditions is `NodeReady`. In the future, we plan to add more.
```
Comment (davidopp) [inline docs/node.md]: s/conditions/condition/

File: docs/node.md [hunk 2]
```diff
@@ -38,24 +38,19 @@ must have appropriate conditions, see below.
...
-`NodeSchedulable` means node is allowed to schedule any new pods and is
-controlled by 'unschedulable' field in node spec. Different condition provides
-different level of understanding for node health. Kubernetes will make a
-comprehensive scheduling decision based on the information. Node condition
-is represented as a json object. For example, the following conditions mean
-the node is in sane state but not allowed to accept new pods:
+conditions is `NodeReady`. In the future, we plan to add more. 
+`NodeReady` means kubelet is healthy and ready to accept pods. Different 
+condition provides different level of understanding for node health. 
+Kubernetes will make a comprehensive scheduling decision based on the
```
Comment (davidopp) [inline docs/node.md]: I would remove this sentence ("Kubernetes will make a comprehensive...") It doesn't make as much sense anymore now that NodeSchedulable is not a condition.

File: docs/node.md [hunk 3]
```diff
@@ -38,24 +38,19 @@ must have appropriate conditions, see below.
...
-different level of understanding for node health. Kubernetes will make a
-comprehensive scheduling decision based on the information. Node condition
-is represented as a json object. For example, the following conditions mean
-the node is in sane state but not allowed to accept new pods:
+conditions is `NodeReady`. In the future, we plan to add more. 
+`NodeReady` means kubelet is healthy and ready to accept pods. Different 
+condition provides different level of understanding for node health. 
+Kubernetes will make a comprehensive scheduling decision based on the 
+information. Node condition is represented as a json object. For example, 
+the following conditions mean the node is in sane state but not allowed to
```
Comment (davidopp) [inline docs/node.md]: this is wrong now that you removed Schedulable. So just delete everything in this sentence starting with "but not allowed..."

File: pkg/api/types.go [hunk 4]
```diff
@@ -1064,12 +1064,10 @@ type NodeConditionType string
 
 // These are valid conditions of node. Currently, we don't have enough information to decide
 // node condition. In the future, we will add more. The proposed set of conditions are:
-// NodeReachable, NodeLive, NodeReady, NodeSchedulable, NodeRunnable.
+// NodeReady
```
Comment (davidopp) [inline pkg/api/types.go]: I think we plan to add NodeReachable back in the future maybe, so I would add that one to the list.

**Comment by davidopp** [comment]
Your NodeController changes for timestamp cache got mixed into this one, but the changes you made for schedulable LGTM.

File: docs/node.md (hunk 1 above)
Comment (gmarek) [inline docs/node.md]: Done.

File: docs/node.md (hunk 2 above)
Comment (gmarek) [inline docs/node.md]: Done.

File: docs/node.md (hunk 3 above)
Comment (gmarek) [inline docs/node.md]: Done.

File: pkg/api/types.go (hunk 4 above)
Comment (gmarek) [inline pkg/api/types.go]: Done.

**Comment by gmarek** [comment]
Sorry, fixed that.

**Comment by gmarek** [comment]
@bgrant0607 It is a bit of work, as I don't know how to easily tell DeepEqual/defaults to ignore NodeSchedulable Condition as long as Unschedulable Spec matches. Do you think it's worth the effort?

**Comment by thockin** [comment]
I have not really followed all the work around this topic, so pardon my dumb questions.  Can't you sync the Unschedulable bit from the status on conversion?

**Comment by gmarek** [comment]
That I did, and I should work "in production". 

The problem is in tests, which don't like having a Schdulable Condition in one object and not having it in the other. I tried modifying defaults.go in v1beta1, but I probably did something wrong or I forgot to do something, as it was not enough.

**Comment by bgrant0607** [comment]
Feel free to punt on the condition in v1beta1/2. I don't know that anyone was using it. It wasn't there for very long, and isn't part of spec.

**Comment by thockin** [comment]
Ah, yeah, I agree with Brian, if you can make that work.

**Comment by gmarek** [comment]
Rebased. @bgrant0607 - what do you mean by 'punt on condition'? That I should just remove them, or not fill them up (so leave this PR as it is)?

**Comment by bgrant0607** [comment]
Please add comments to `v1beta[12]/types.go` that indicate that NodeSchedulable is deprecated and not populated. Otherwise, this is fine.

**Comment by gmarek** [comment]
Done.

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
The shippable failure looks unrelated.

**Pull request merged**